### PR TITLE
Implement voting on forum posts

### PR DESCRIPTION
**Pull request by lowjoel**
Implements #868. Depends on #993.

### CODE REVIEW

File: app/models/concerns/course/discussion/post/ordering_concern.rb [hunk 1]
```diff
@@ -41,18 +41,10 @@ def sort(post_id)
...
-  end
-
-  included do
-    scope :ordered_topologically, (lambda do
-      ScopeMethods.instance_method(:ordered_topologically).bind(current_scope).call
-    end)
+  # Returns a set of recursive arrays indicating the parent-child relationships of posts.
+  #
+  # @return [Enumerable]
+  def ordered_topologically
```
Comment (allenwq) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: @lowjoel This fix is still needed ?

Comment (lowjoel) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: It's not really a fix... it's really the method that needs to be used.

Comment (allenwq) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: but you defined it as a scope before which also can be used  ?

Comment (lowjoel) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: I removed the scope, it's now just a method. In the Post class it's a class method, in the topic class it's an association method...

Comment (allenwq) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: So it servers same functionality as the original scope which you removed right ?

Comment (lowjoel) [inline app/models/concerns/course/discussion/post/ordering_concern.rb]: yes, otherwise the eager loading will be lost.

**Pull request merged**